bot/core/paginator.py

```diff
@@ -0,0 +1,421 @@
+
+        for index, arg in enumerate(args):
+            if isinstance(arg, Embed):
+                self.embeds.append(arg)
+            elif isinstance(arg, list):
+                _embed = discord.Embed()
+                for element in arg:
+                    if isinstance(element, dict):
+                        # If element is an empty dictionary
+                        if element == {}:
+                            raise PaginatorError(
+                                "FieldPaginator argument is a nested dictionary but does not have the mandatory "
+                                "key-value pair".strip()
+                            )
+                        elif len(element.items()) > 2:
+                            raise PaginatorError(
+                                "FieldPaginator argument is supposed to be a dictionary of "
+                                "2 key-value pair (key: value, inline: bool), but found " + len(element.items())
+                            )
+
+                        _embed.add_field(
+                            name=list(element.keys())[0],
+                            value=list(element.values())[0],
+                            inline=True if 'inline' not in element.keys() else bool(element['inline'])
+                        )
+                    else:
+                        raise PaginatorError(
+                            "\n"
+                            "Field paginator nested embed is not accepted\n"
+                            "Eg: FieldPaginator([{'Name': 'John', 'inline': False}, ->Embed().set_footer('Name', 'Sarah')<-])\n"
+                            "Accepted type to create an embed for FieldPaginator (note that this will create a new page):\n"
+                            "FieldPaginator([{'Name': 'John', 'inline': False}], Embed().set_footer('Name', 'Sarah'))"
+                        )
+                self.embeds.append(_embed)
+            else:
+                raise PaginatorError(
+                    "FieldPaginator argument given "
+                    f"at index {index} is invalid.\n"
+                    "The supported types are:\n"
+                    "Embed and list"
+                )
+
+
+class ListPaginator(_BasePaginator):
+    """
+    A class which creates a list of embeds and configures
+    their descriptions according to the parameters.
+
+    Parameters
+    ----------
+        entries : List[str]
+            List of entries to add to the paginator
+            Length of every entry must be lower than
+            the max_size argument.
+        individual_paging : bool
+            Whether to make a new paginator page for every
+            entry.
+        prefix : str
+            A string that is added to the beginning to every
+            paginator page.
+        suffix : str
+            A string that is added to the end of every paginator
+            page.
+        max_size : int
+            Maximum amount of characters allowed per entry.
```

Comment:
This should be a docstring for `__init__`, not for the class itself

**This also applies to all occurrences of this issue**